File: backend/java/src/main/java/de/iits/elocalculatorbackend/match/service/MatchService.java
package de.iits.elocalculatorbackend.match.service;

import de.iits.elocalculatorbackend.match.model.dto.MatchRequestDto;
import de.iits.elocalculatorbackend.match.model.dto.MatchResponseDto;
import de.iits.elocalculatorbackend.match.model.entity.Match;
import de.iits.elocalculatorbackend.match.repository.MatchRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class MatchService {

    private MatchRepository repository;

    public List<MatchResponseDto> getAllMatches(){
        return repository.findAll().stream().map(MatchResponseDto::new).toList();
    }

    public MatchResponseDto createMatch(MatchRequestDto createMatch) {
        Match saved = repository.save(createMatch.toEntity());
        return new MatchResponseDto(saved);
    }
}

File: backend/springboot-kotlin/src/main/kotlin/de/iits/elo/match/service/MatchService.kt
package de.iits.elo.match.service

import de.iits.elo.match.model.dto.MatchRequestDto
import de.iits.elo.match.model.dto.MatchResponseDto
import de.iits.elo.match.model.dto.toEntity
import de.iits.elo.match.model.entity.Match
import de.iits.elo.match.model.entity.toResponseDto
import de.iits.elo.match.repository.MatchRepository
import org.springframework.stereotype.Service

@Service
class MatchService(
    private val matchRepository: MatchRepository,
) {

    fun findAll(): List<MatchResponseDto> =
        matchRepository.findAll().map(Match::toResponseDto)

    fun create(match: MatchRequestDto): MatchResponseDto =
        matchRepository.save(match.toEntity()).toResponseDto()
}
